docs/rules/max-lines-per-function.md

```diff
@@ -0,0 +1,130 @@
+# enforce a maximum function length (max-lines-per-function)
+
+Some people consider large functions a code smell. Large functions tend to do a lot of things and can make it hard following what's going on. While there is not an objective maximum number of lines considered acceptable in a function, most people would agree it should not be above 200. Google's JS style guide (by proxy of their C++ style guide), recommends 40 lines or fewer per function. The Linux Kernel's style guide recommends 48 lines or fewer.
+
+## Rule Details
+
+This rule enforces a maximum number of lines per function, in order to aid in maintainability and reduce complexity. This rule does not count any top-level function code.
+
+
+## Options
+
+This rule has a number or object option:
```

Comment:
200 is a bit pointlessly high...if you're writing a function *that* long, you're doing something *massively* wrong 99% of the time. (The remaining 1% is for things like config files and autogenerated code.)

I could see 50 as a decent default. For comparison, Rubocop's [`Metrics/MethodLength`](http://rubocop.readthedocs.io/en/latest/cops_metrics/#metricsmethodlength) defaults to 10 (which is *pretty* low, but still workable).